### PR TITLE
Fix #41 

### LINKED ISSUE #41: Add more images to existing products
Add a button on the product page to add fresher or missing images to an existing product.
Here's a mockup of how it would ideally work: a button on each tab, with a language dropdown. It allows the user to direct an image to a function (ingredient/nutrition/front) and a language (product can have up to 20 languages)

The current photo of the ingredients (for example) would be display, or an informative text (There is no photo of ingredients in French)

image

A button in each tab to add a photo if no picture is available 
- one in the main tab for the front image and additional images
- one button for the ingredient photo
- one for the nutrition photo

Behaviour: 
- if a front/ingredients/nutrition image is already selected for this language, show "Add more images" and upload images without selecting them
- if no front/ingredients/nutrition image is selected for this language, show "Add images" and upload images without selecting them (the server will do the job)

### PR DESCRIPTION
**Pull request by itchix**
First version

### CODE REVIEW

**Comment by itchix** [comment]
@herau do you have time to review it?

**Comment by teolemon** [comment]
## Bugs
- The photos are uploaded but not selected in some cases (probably has to do with languages on preexisiting products) (Quick fix: assign the language of the product)

## Small fixes
- the background of the Add button should be the same as the page it's on
- the Take a picture text is either below or above the icon. Put it always below.
- Rename the buttons: 

1. **Take a front picture**
2. **Take a picture of the ingredients**
3. **Take a picture of the nutrition table**

- Add a **Take more photos** button in the _Summary_ tab

## Expanding the feature for more usecases

- Add the option to use the offline upload method either when no network is found or when an option is selected in the settings for people who don't want to upload immediately on Mobile (they have a metered 3G connexion, they are abroad…)
- No language of the image support

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/ProductImageField.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * Kind of Product Image
  */
 public enum ProductImageField {
-    FRONT, INGREDIENTS, NUTRITION;
+    FRONT, INGREDIENTS, NUTRITION, OTHER;
```
Comment (herau) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/ProductImageField.java]: le type correspond à l'api front, ingredient et nutrition. 
tu es sur que l'API attend un autre type `other` ?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java [hunk 2]
```diff
@@ -353,7 +356,17 @@ public void postImg(final Context context, final ProductImage image) {
         lt.setTextColor(context.getResources().getColor(R.color.white));
         lt.show();
 
-        apiService.saveImage(image.getCode(), image.getField(), image.getImguploadFront(), image.getImguploadIngredients(), image.getImguploadNutrition())
+        String lang = Locale.getDefault().getLanguage();
+
+        HashMap<String, RequestBody> imgMap = new HashMap<>();
```
Comment (herau) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java]: use a generic interface for declaration like `Map<String, RequestBody> imgMap = new HashMap<>();`

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -368,5 +368,9 @@
     <string name="collagen_meat_protein_ratio">Collagen meat protein ratio</string>
     <string name="cocoa">Cocoa</string>
     <string name="chlorophyl">Chlorophyl</string>
+    <string name="take_picture_ingredients">Take a picture of the ingredients</string>
+    <string name="take_picture_front">Take a picture of the front</string>
+    <string name="take_picture_nutrition_table">Take a picture of the nutrition table</string>
+    <string name="take_more_pictures">Take more pictures of any interesting information</string>
```
Comment (herau) [inline app/src/main/res/values/strings.xml]: j'aurai mis uniquement `take more pictures`

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/ProductImageField.java (hunk 1 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/ProductImageField.java]: @stephanegigandet approved ;-)

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: ok for "Take more pictures"

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java (hunk 2 above)
Comment (itchix) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java]: Ok

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/ProductImageField.java (hunk 1 above)
Comment (itchix) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/ProductImageField.java]: Yes =)

**Comment by itchix** [comment]
close & fix #41

**Pull request merged**